source/opcode.cpp

```diff
@@ -417,3 +417,22 @@ bool spvOpcodeIsBlockTerminator(SpvOp opcode) {
   return spvOpcodeIsBranch(opcode) || spvOpcodeIsReturn(opcode) ||
          opcode == SpvOpKill || opcode == SpvOpUnreachable;
 }
+bool spvOpcodeIsOpaqueType(SpvOp opcode) {
```

Comment:
According to the spec an opaque-ness is transitive. Any type containing or pointing to or containing a pointer to an opaque type is also opaque. Should this function instead take an instruction and recurse through its types?

Reply:
I'll rename this function to `spvOpcodeIsBaseOpaqueType`.  It cannot take an instruction because the file is shared with the validator, and instructions are represented differently.

I think I will need something more generic for the check that currently calls this.

Reply:
I've add `IsOpaqueType` to the instruction class.  That will look re